extension/src/service/web-component-service.js

```diff
@@ -13,33 +13,48 @@ function getPaymentMethods(merchantAccount, getPaymentMethodsRequestObj) {
   )
 }
 
-function makePayment(merchantAccount, makePaymentRequestObj) {
+function makePayment(
+  merchantAccount,
+  commercetoolsProjectKey,
+  makePaymentRequestObj
+) {
   const adyenCredentials = config.getAdyenConfig(merchantAccount)
+  extendRequestObjWithMetadata(makePaymentRequestObj, commercetoolsProjectKey)
```

Comment:
web-component-service.js getting uglier with every change, maybe it's to time to restructure it? Let me know if you want to discuss making it cleaner. Might be another PR for it?

Reply:
Not in diff but looks like an issue in the validation logic, this should be an invalid request if custom.fields.merchantAccount and request.merchantAccount, right? 
```
if (!requestObj.merchantAccount)
    requestObj.merchantAccount = adyenMerchantAccount
```

Reply:
> web-component-service.js getting uglier with every change, maybe it's to time to restructure it? Let me know if you want to discuss making it cleaner. Might be another PR for it?

We can discuss it, but I'm not sure when and how. This extending thing is just for some requests, so it's not very easy to generalize.

Reply:
> Not in diff but looks like an issue in the validation logic, this should be an invalid request if custom.fields.merchantAccount and request.merchantAccount, right?
> 
> ```
> if (!requestObj.merchantAccount)
>     requestObj.merchantAccount = adyenMerchantAccount
> ```

I tried to remove it, but for example for Cancellation or Refund this is valid code. For them we create a request from a transaction and in transaction we don't have `merchantAccount`. One way is to add this part only in the methods that use it. Then it will cause less confusion, but we have some code duplication. WDYT is better?

Reply:
> We can discuss it, but I'm not sure when and how. This extending thing is just for some requests, so it's not very easy to generalize.

Let me check it, we could discuss this in a different PR.. For now, pass my review comment.

Reply:
My concerns not for the place that we assign merchant account to request object (i.e: makePayment, cancel and refund request), here when custom.fields.adyenMerchantAccount is different than requestObj.merchantAccount we should not allow sending it to adyen because we can not match notification for it.. I wonder if we could add this validation before sending a request.

Reply:
But if it's different, Adyen will reject such request. We use `custom.fields.adyenMerchantAccount` to get an adyen config and then we send the request to it. If request contains a merchantAccount that does not exist, it will fail immediately and it will not go to notification.

Reply:
for instance:  is it possible to send a payment for the `merchant2` when the `custom.fields.adyenMerchantAccount` is set to `merchant1`, in case both `merchant1` and `merchant2` is valid and configured?

Reply:
It's possible to send. But Adyen will reject it I guess. Let's take it step by step:

1. `custom.fields.adyenMerchantAccount` is used to load the adyen config from the configuration. In this case we will load configuration for `merchant1`
2. In the request, we have `merchant2` set as `merchantAccount` and we will send this `merchant1` account.

So I hope in this case adyen will reject it :) But it's true we did not test this case as we don't have 2 separate accounts. Our "accounts" for testing have the same clientKey and apiKey, so I guess it's just a namespace in the same adyen account. We can test it when we get real 2 accounts from Adyen.

Reply:
For me, this looks like a validation issue when you have the same API client for different merchant accounts (which is possible by Adyen) and the module should not allow the payments when there is a difference in the metadata and make payment.. but will not block the PR just for this, as you suggested please handle the testing when you have the accounts.